src/main/java/frc/robot/poseestimator/limelights/SmartLimelights.java

```diff
@@ -0,0 +1,114 @@
+package frc.robot.poseestimator.limelights;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import frc.utils.GBSubsystem;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Optional;
+
+
+public class SmartLimelights extends GBSubsystem {
+
+	private List<Limelight> limelights;
+	private LimelightsHardware limelightHardware;
+
+	public SmartLimelights(String[] limelightNames) {
+		super(VisionConstants.DEFAULT_CONFIG.Logpath());
+
+		this.limelightHardware = new LimelightsHardware(limelightNames, VisionConstants.DEFAULT_CONFIG.HardwareLogpath());
+		this.limelights = limelightHardware.getAllLimelights();
+	}
+
+	public List<Optional<Pair<Pose2d, Double>>> getAll2DEstimates() {
+		ArrayList<Optional<Pair<Pose2d, Double>>> estimates = new ArrayList<>();
+
+		for (Limelight limelight : limelights) {
+			if (!filterOutLimelight(limelight)) {
+				estimates.add(limelight.getUpdatedPose2DEstimation());
+			}
+		}
+
+		return estimates;
+	}
```

Comment:
why not in "limelights hardware"?

Reply:
This is the filtered version of the limelights. LimelightsHardware represents the bare hardware, while this method preforms smart filtering. I'll change it's name to make this clear.

Reply:
Renamed to getAllFiltered... or smth. It's smart filtering, and doesn't represent the hardware.

Reply:
it's dead for 3 days so mark as resolved